File: 3653-xor-after-range-multiplication-queries-i/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int xorAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
        const long long MOD = 1e9 + 7;
        int n = nums.size();

        // Process each query sequentially
        for (const auto& query : queries) {
            int l = query[0];
            int r = query[1];
            int k = query[2];
            long long v = query[3];

            // Perform range multiplication with step k
            for (int i = l; i <= r; i += k) {
                nums[i] = (1LL * nums[i] * v) % MOD;
            }
        }

        // Calculate bitwise XOR of all elements in the modified array
        int result = 0;
        for (int x : nums) {
            result ^= x;
        }

        return result;
    }
};

File: 3653-xor-after-range-multiplication-queries-i/solution.py
class Solution:
    def xorAfterQueries(self, nums: list[int], queries: list[list[int]]) -> int:
        MOD = 10**9 + 7
        n = len(nums)
        
        # Process each query: [l, r, k, v]
        for l, r, k, v in queries:
            # Update nums[idx] starting from l, up to r, jumping by k
            for idx in range(l, r + 1, k):
                nums[idx] = (nums[idx] * v) % MOD
        
        # Compute the final bitwise XOR sum of all elements
        xor_sum = 0
        for val in nums:
            xor_sum ^= val
            
        return xor_sum